rust/agama-server/src/web/ws.rs

```diff
@@ -50,45 +49,49 @@ pub async fn ws_handler(
     Extension(client_id): Extension<Arc<ClientId>>,
     ws: WebSocketUpgrade,
 ) -> impl IntoResponse {
-    ws.on_upgrade(move |socket| handle_socket(socket, state.events, state.old_events, client_id))
+    ws.on_upgrade(move |socket| handle_socket(socket, state.events, client_id))
 }
 
-async fn handle_socket(
-    mut socket: WebSocket,
-    events: event::Sender,
-    old_events: http::event::OldSender,
-    client_id: Arc<ClientId>,
-) {
+async fn handle_socket(mut socket: WebSocket, events: event::Sender, client_id: Arc<ClientId>) {
     let mut events_rx = events.subscribe();
-    let mut old_events_rx = old_events.subscribe();
 
     let conn_event = agama_lib::event!(ClientConnected, client_id.as_ref());
     if let Ok(json) = serde_json::to_string(&conn_event) {
-        _ = socket.send(Message::Text(json)).await;
+        if socket.send(Message::Text(json)).await.is_err() {
+            return;
+        }
     }
 
     loop {
         tokio::select! {
-            msg = old_events_rx.recv() => {
-                if let Err(e) = send_msg(&mut socket, msg).await {
-                    eprintln!("Error sending old event: {e:?}");
+            // Handle messages from the client
+            msg = socket.recv() => {
+                match msg {
+                    Some(Ok(Message::Close(_))) | Some(Err(_)) | None => break,
```

Comment:
Maybe this deserve a comment as I need to check exact websocket API to understand all cases (explicit close message, None for closed stream and error for generic axum error ). And I am not sure if breaking in case of generic axum error is good or not as sadly reading axum code for method `recv` does not help understand when it can return Err